File: fractions/tests/test_fraction/test_functions.py
def test_arithmetic_add(f1: Fraction, f2: Fraction, f3: Fraction, test_n: int):
    if f3 == f1 + f2:
        result = "\033[92mPassed\033[0m"
    else:
        result = "\033[91mFailed\033[0m"
    tm1 = f"Test #{test_n}"
    tm2 = str(f1)
    tm3 = str(f2)
    tm4 = str(f3)
    tm5 = str(f1 + f2)
    test_result = f"{tm1} {tm2:<6} + {tm3:<6} : exp {tm5:<12} : got {tm4:<12} : {result}"

test_list = [(1, test_initialization),
             (2, test_isproper),
             (3, test_isimproper),
             (4, test_arithmetic_add),
             (5, test_arithmetic_sub),
             (6, test_arithmetic_mul),
             (7, test_arithmetic_div),
             (8, test_assignment),
             (9, test_assignment_add),
             (10, test_assignment_sub),
             (11, test_assignment_mul),
             (12, test_assignment_div),
             (13, test_compare_equal),
             (14, test_compare_unequal),
             (15, test_compare_lesser),
             (16, test_compare_bigger),
             (17, test_compare_lequal),
             (18, test_compare_bequal)]

File: fractions/tests/test_fraction/cpp/src/test_functions.cpp
#include "../inc/test_fraction.hpp"

void	test_initialization() {
	Fraction f1 {0, 1};
	cin >> f1;
	cout << f1;
}

void	test_isproper() {
	Fraction f1 {0, 1};
	cin >> f1;
	cout << f1.isProper();
}

void	test_isimproper() {
	Fraction f1 {0, 1};
	cin >> f1;
	cout << f1.isImproper();
}

void	test_arithematic_add() {
	Fraction f1 {0, 1}, f2 {0, 1};
	cin >> f1 >> f2;
	cout << f1 + f2;
}

